Sources/AngouriMath/Functions/Simplification/Patterns/Patterns.Common.cs

```diff
@@ -227,6 +227,16 @@ internal static partial class Patterns
             Mulf(Rational(var mOne, var den), var any1) when mOne == -1 && den != 1 => -(any1 / den),
             Mulf(var any1, Rational(var mOne, var den)) when mOne == -1 && den != 1 => -(any1 / den),
 
+            // a xor false = a
+            Xorf(var any1, Entity.Boolean any2) when any2 == false => any1,
+            Xorf(var any2, var any1) when any2 == false => any1,
+            Xorf(var any1, Integer any2) when any2 == 0 => any1,
+            Xorf(Integer any2, var any1) when any2 == 0 => any1,
```

Comment:
There's no logic backing xor on *numbers*, so this pattern is not particularly useful. Or you have some other, broader idea regarding xor? Please, share!

Reply:
I'm not sure what you mean about backing logic; are you referring to the solver system? If so, we're just using the symbolic algebra representation and simplification systems so this isn't something we had considered. In quantum notation, you see things like this quite frequently:

![image](https://user-images.githubusercontent.com/627896/130550875-6ff50502-13dd-4264-9f0f-72db81c29a79.png)

That is, some variable XOR'd against some other variable. In some cases one of those variables is simply the integer 0, and that can be simplified to just using the other value. That's the use case this was intended to support.

Reply:
Right. What I am saying is that we don't support boolean operators on integers. So if we want to add it, it should be added for all boolean operators and for all cases, not only for a few cases for `xor`. That's what I mean 😅

Reply:
By the way, maybe you want to join our discord server (see the repo's readme's green badge)? It's a chat, so we can discuss it more conveniently there (and having others potentially involved)

Reply:
Sure, I'll hop in tomorrow if you believe there's more to this discussion that what you've mentioned here.